blog/2019-12-18-HowToPlanAndOptimizeASecretSanta.adoc

```diff
@@ -0,0 +1,102 @@
+= How to Plan (and Optimize) a Secret Santa
+:page-interpolate: true
+:awestruct-author: cchianel
+:awestruct-layout: blogPostBase
+:awestruct-tags: [useCase]
+
+Many businesses host a "Secret Santa", where each employee get assigned
+a coworker whom they need to buy a present for. This fosters good relations
+between the employees and bring them closer together. But what about global
+companies, where coworkers may be many miles apart? Ideally, we want
+employees who are further apart to give gifts to each other, since they
+are the ones who probably know the least about each other. Let optimize
+it with OptaPlanner!
+
+== The Constraints
+
+One of the most obvious constraint is that everyone should get a gift.
+Or phrased a little differently: no one should get multiple gifts.
+
+Here is how one would create this constraint in OptaPlanner using the
+constraint streams API:
+[source,java]
+----
+private Constraint sameRecieverConflict(ConstraintFactory constraintFactory) {
+    return constraintFactory
+            .fromUniquePair(SecretSantaAssignment.class,
+                    Joiners.equal(SecretSantaAssignment::getReciever))
+            .penalize("Same Reciever", HardMediumSoftBigDecimalScore.ONE_HARD);
+}
+----
+
+Next, we need to to award assignments where the gifter and the reciever
+are farther apart. Here how we will do it with constraint streams:
+[source,java]
+----
+private Constraint largerDistanceAward(ConstraintFactory constraintFactory) {
+    return constraintFactory
+            .from(SecretSantaAssignment.class)
+            .rewardConfigurableBigDecimal("secretFactor", "Larger Distance Award",
+                              (m) -> BigDecimal.valueOf(Location.calculateDistanceBetween(m.getGifter().getLocation(),
+                                                                       m.getReciever().getLocation())));
+}
+----
+
+Let try solving with just these two constraints to see what we get:
+
+image::noPairChecking.png[]
+
+Alice is assigned to Daniella, Daniella is assigned to Alice...
+Bob is assigned to Austin, Austin is assigned to Bob...
+Something about this solution seems odd; it only generate assignments
```

Comment:
`it only generates assignments...`

Reply:
Subject-verb agreement - why does it exist....

Reply:
/me realizes it is the subject, not assigments, although what is it; generates probably be changed to has